### ISSUE TITLE
abuse of spinlock

### ISSUE THREAD

**Issue by cocalele** [cocalele/PureFlash#22]
Too many spinlocks, 
1) in PfFixedSizeQueue , there's spinlock protected.
2) in PfEventQueue, there's also a spinlock.

each time `post_event` is called, the two spinlock will be both acquired.
```
int PfEventQueue::post_event(int type, int arg_i, void* arg_p, void* arg_q)
{
	//S5LOG_INFO("post_event %s into:%s", EventTypeToStr((S5EventType)type), name);
	{
		AutoSpinLock _l(&lock);
		int rc = current_queue->enqueue(S5Event{ type, arg_i, arg_p , arg_q});
		if(rc)
			return rc;
	}
	write(event_fd, &event_delta, sizeof(event_delta));
	return 0;
}
```

**Comment by cocalele** [comment]
```
	U* alloc() {
		AutoSpinLock _l(&lock);
		if (free_obj_queue.is_empty())
			return NULL;
		return free_obj_queue.dequeue(); //TODO: double locked here!!
	}
```

**Comment by cocalele** [comment]
Has fixed in commit bb78c76d
